plugins/window_size/lib/src/window_size_channel.dart

```diff
@@ -113,12 +128,53 @@ class WindowSizeChannel {
   void setWindowFrame(Rect frame) async {
     assert(!frame.isEmpty, 'Cannot set window frame to an empty rect.');
     assert(frame.isFinite, 'Cannot set window frame to a non-finite rect.');
-    try {
-      await _platformChannel.invokeMethod(_setWindowFrameMethod,
-          [frame.left, frame.top, frame.width, frame.height]);
-    } on PlatformException catch (e) {
-      print('Platform exception setting window frame: ${e.message}');
-    }
+    await _platformChannel.invokeMethod(_setWindowFrameMethod,
+        [frame.left, frame.top, frame.width, frame.height]);
+  }
+
+  /// Sets the minimum size of the window containing this Flutter instance.
+  void setWindowMinSize(Size size) async {
+    await _platformChannel
+        .invokeMethod(_setWindowMinimumSizeMethod, [size.width, size.height]);
+  }
+
+  // Window maximum size unconstrained is passed over the channel as -1.
+  double _channelRepresentationForMaxDimension(double size) {
+    return size == double.infinity ? -1 : size;
+  }
+
+  /// Sets the maximum size of the window containing this Flutter instance.
+  void setWindowMaxSize(Size size) async {
+    await _platformChannel.invokeMethod(_setWindowMaximumSizeMethod, [
+      _channelRepresentationForMaxDimension(size.width),
+      _channelRepresentationForMaxDimension(size.height),
+    ]);
+  }
+
+  /// Gets the minimum size of the window containing this Flutter instance.
+  Future<Size> getWindowMinSize() async {
+    final response =
+        await _platformChannel.invokeMethod(_getWindowMinimumSizeMethod);
+    return _sizeFromWHList(
+        List<double>.from(response[_windowSizeKey].cast<double>()));
```

Comment:
If there's only one entry then there's no need for a map, and this can just return the list directly (and I'm not seeing any reason there would be anything more in the future, nor does the setter use this extra layer). Then you can use  [invokeListMethod](https://api.flutter.dev/flutter/services/MethodChannel/invokeListMethod.html) to avoid the explicit casting here as well.

Same for max size.

Reply:
Done